### PR TITLE
Migrated further widgets to ReactJS

### PR DESCRIPTION
**Pull request by jbicker**
In this PR I migrated the following widgets:
Output
ExtensionDetail
GitNavigable
GitDiff
GitHistory
PluginExt

### CODE REVIEW

File: packages/git/src/browser/diff/git-diff-widget.tsx [hunk 1]
```diff
@@ -168,70 +171,74 @@ export class GitDiffWidget extends GitNavigableListWidget<GitFileChangeNode> imp
...
         });
     }
 
-    protected renderNavigationLeft(): h.Child {
-        return h.span({
-            className: "fa fa-arrow-left",
-            title: "Previous Change",
-            onclick: () => this.navigateLeft()
-        });
+    protected readonly previousChange = () => this.doPreviousChange();
```
Comment (akosyakov) [inline packages/git/src/browser/diff/git-diff-widget.tsx]: fyi: method names should be verbs, i.e. `showPreviousChange`

File: packages/git/src/browser/git-navigable-list-widget.ts [hunk 2]
```diff
@@ -14,20 +14,22 @@
...
 import { inject, injectable } from "inversify";
+import { ReactWidget } from "@theia/core/lib/browser/widgets/react-widget";
 
 @injectable()
-export class GitNavigableListWidget<T extends { selected?: boolean }> extends VirtualWidget {
+export abstract class GitNavigableListWidget<T extends { selected?: boolean }> extends ReactWidget {
 
     protected gitNodes: T[];
     private _scrollContainer: string;
+    protected scrollArea: HTMLElement;
```
Comment (akosyakov) [inline packages/git/src/browser/git-navigable-list-widget.ts]: where is it assigned?

File: packages/git/src/browser/git-navigable-list-widget.ts [hunk 3]
```diff
@@ -52,12 +54,10 @@ export class GitNavigableListWidget<T extends { selected?: boolean }> extends Vi
 
     protected onUpdateRequest(msg: Message): void {
         super.onUpdateRequest(msg);
-
         (async () => {
             const selected = this.node.getElementsByClassName(SELECTED_CLASS)[0];
-            const scrollArea = await this.getScrollContainer();
-            if (selected && scrollArea) {
-                ElementExt.scrollIntoViewIfNeeded(scrollArea, selected);
+            if (selected && this.scrollArea) {
```
Comment (akosyakov) [inline packages/git/src/browser/git-navigable-list-widget.ts]: `this.scrollArea` does not have  `undefined` in type declaration, why is it checked?

File: packages/git/src/browser/git-navigable-list-widget.ts (hunk 2 above)
Comment (jbicker) [reply packages/git/src/browser/git-navigable-list-widget.ts]: nowhere. was a relict of ancient times. sorry.

**Comment by akosyakov** [comment]
When I scroll in the git history I get a lot of errors:
screen shot 2018-07-12 at 10 14 57

**Comment by akosyakov** [comment]
Get such errors if I reload with the git history opened:
screen shot 2018-07-12 at 10 16 44

**Comment by jbicker** [comment]
There are already follow up issues. The GitHistory was already broken before. 
https://github.com/theia-ide/theia/issues/2296
https://github.com/theia-ide/theia/issues/2295
https://github.com/theia-ide/theia/issues/2294

**Pull request merged**